### PR TITLE
Upgrade to es6

### PR DESCRIPTION
**Pull request by montogeek**
This PR upgrades code to ES6.
All tests pass.
Added EditorConfig and ESLint configs to make it easier for future contributions.

`createWriteStream` accepted a second parameter called `options`, I removed since it is not being used, although Node implementation uses it: https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options https://github.com/nodejs/node/blob/master/lib/fs.js#L1878

### CODE REVIEW

**Comment by TheLarkInn** [comment]
Hey @montogeek would you mind syncing with master and then check tests again. Since we just dropped 0.12, I think its time to dust this off and get to work. https://github.com/webpack/webpack/pull/3605

**Comment by montogeek** [comment]
@TheLarkInn Synced, tests pass 💃

File: lib/MemoryFileSystem.js [hunk 1]
```diff
@@ -3,28 +3,25 @@
...
-	if (Error.captureStackTrace)
-		Error.captureStackTrace(this, arguments.callee)
-	this.code = err.code;
-	this.errno = err.errno;
-	this.message = err.description;
-	this.path = path;
+class MemoryFileSystemError extends Error {
+	constructor(err, path) {
+		super(err, path);
+		if (Error.captureStackTrace)
```
Comment (wtgtybhertgeghgtwtg) [inline lib/MemoryFileSystem.js]: Isn't this always the case for `node >= 4`?

**Comment by montogeek** [comment]
Sure!

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,7 @@
 language: node_js
 node_js:
-  - "0.10"
-  - "0.12"
-  - "iojs"
+  - "4"
+  - "6"
```
Comment (sokra) [inline .travis.yml]: also a node 7

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,220 @@
+{
+    "env": {
+        "node": true,
+				"es6": true
```
Comment (sokra) [inline .eslintrc.json]: spacing incorrect in this line

**Comment by montogeek** [comment]
Coveralls says that the coverage decreased, but those lines were already uncovered in `master` branch

**Comment by TheLarkInn** [comment]
Looks good to me @d3viant0ne will likely come back around to update the Travis file to conform to standards. Thank you very much!!

**Comment by montogeek** [comment]
Thanks @TheLarkInn

**Pull request merged**